### PR TITLE
Update rubocop, support files to new ext std

### PR DESCRIPTION
**Pull request by fastjames**
Update rubocop and other support files to match the new standard laid
out by `solidus_dev_support` and implement in upstream master.

### CODE REVIEW

**Comment by fastjames** [comment]
This is now working _and_ completes in less than an hour in CI.

File: spec/controllers/spree/admin/product_catalogs_controller_spec.rb [hunk 1]
```diff
@@ -10,7 +12,7 @@
   stub_authorization!
 
   describe 'GET #index' do
-    it 'should reuturns 200 status' do
+    it 'reuturnses 200 status' do
```
Comment (cnorm35) [inline spec/controllers/spree/admin/product_catalogs_controller_spec.rb]: typo, unless this was written by Gollum

File: solidus_product_feed.gemspec [hunk 2]
```diff
@@ -1,38 +1,30 @@
...
 
 Gem::Specification.new do |s|
   s.platform    = Gem::Platform::RUBY
   s.name        = 'solidus_product_feed'
   s.version     = SolidusProductFeed::VERSION
-  s.summary     = 'Solidus extension that provides an RSS feed for products'
-  s.description = 'A Solidus extension that provides an RSS feed for products'
-  s.required_ruby_version = '>= 1.8.7'
+  s.summary     = 'A Spree extension that provides an RSS feed for products'
+  s.description = 'A Spree extension that provides an RSS feed for products, with Google Shopper extensions'
```
Comment (cnorm35) [inline solidus_product_feed.gemspec:13]: any reason this is being updated to `Spree` over `Solidus`, something matching upstream?

File: app/services/spree/product_feed_service.rb [hunk 3]
```diff
@@ -71,11 +68,7 @@ def google_product_category
 
     def mpn
       mpn = property_value_by_name('MPN').to_s.downcase
-      if mpn.blank?
-        variant.sku
-      else
-        mpn
-      end
+      mpn.presence || variant.sku
```
Comment (cnorm35) [inline app/services/spree/product_feed_service.rb:71]: TIL about `presence` method.
https://guides.rubyonrails.org/active_support_core_extensions.html#presence

File: Rakefile [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
+  RuboCop::RakeTask.new
 
-RSpec::Core::RakeTask.new
+  task default: %i(first_run rubocop spec)
+rescue LoadError # rubocop:disable Lint/HandleExceptions
+  # no rspec available
+end
 
-task :default do
+task :first_run do # rubocop:disable Rails/RakeEnvironment
```
Comment (cnorm35) [inline Rakefile:19]: with this change, will we need to invoke `bundle exec rake first_run` to prepare the dummy app and everything else that's required?

**Review by cnorm35 — APPROVED** [review]
looks good, nothing blocking merging just had a few clarifying questions and comments

File: spec/controllers/spree/admin/product_catalogs_controller_spec.rb (hunk 1 above)
Comment (fastjames) [reply spec/controllers/spree/admin/product_catalogs_controller_spec.rb]: MY PRECIOUS. Fixed.

File: solidus_product_feed.gemspec (hunk 2 above)
Comment (fastjames) [reply solidus_product_feed.gemspec:13]: That's it, matching upstream.

File: Rakefile (hunk 4 above)
Comment (fastjames) [reply Rakefile:19]: No need, the new `default` task includes `first_run`.

File: app/services/spree/product_feed_service.rb (hunk 3 above)
Comment (k1bs) [reply app/services/spree/product_feed_service.rb:71]: WHOA. That's awesome.

File: app/controllers/spree/admin/product_catalog_images_controller.rb [hunk 5]
```diff
@@ -11,7 +13,7 @@ def update
           else
             flash[:error] = t('spree.admin.product_feed_image.error_uploaded')
           end
-        rescue
+        rescue StandardError
```
Comment (k1bs) [inline app/controllers/spree/admin/product_catalog_images_controller.rb:16]: 💯

File: app/models/spree/product_catalog.rb [hunk 6]
```diff
@@ -24,7 +26,8 @@ def all_selected?(variants)
 
     def populate_default_items
       variants = Spree::Product.all.map(&:variants_including_master).flatten
-      update_attribute(:item_ids, variants.map(&:id).map(&:to_s))
+      variant_ids = variants.map { |pv| pv.id.to_s }
+      update(item_ids: variant_ids)
```
Comment (k1bs) [inline app/models/spree/product_catalog.rb:30]: Could you give some thought as to this change to `update`? My understanding was that its main benefit was for return values. Are we expecting the return value of this method to change?

**Review by k1bs — APPROVED** [review]
Overall, :cake:. Just one question.

File: app/models/spree/product_catalog.rb (hunk 6 above)
Comment (fastjames) [reply app/models/spree/product_catalog.rb:30]: The change is twofold:

1. Change the 2-pass maps into one map
2. Don't skip validation (I'm not sure why we were doing so before).

Comment (fastjames) [reply app/models/spree/product_catalog.rb:30]: I don't recall the exact reasoning for switching from `update_attribute` to `update` but I don't think there is a good reason to skip validations at this point in the code.

Comment (k1bs) [reply app/models/spree/product_catalog.rb:30]: Oh, that makes sense. I actually didn't realize it was `update_attribute` without the `s`.

Thanks.

**Pull request merged**